docs/source/concepts/bento.rst

```diff
@@ -763,30 +763,216 @@ Docker Template (Danger Zone)
 """""""""""""""""""""""""""""
 
 The :code:`docker_template` field gives user the full control over how the
-:code:`Dockerfile` was generated in a Bento.
+:code:`Dockerfile` was generated in a Bento. Users can use this field to
+customize Bento's Dockerfile instruction set to suits their needs.
 
-Documenting this option is working-in-progress, see :issue:`2497`
+Let say that you want to isolate the installation of a custom python library from a build stage to the 
+Bento utilizing `dockerfile's network modes <https://github.com/moby/buildkit/blob/master/frontend/dockerfile/docs/syntax.md#network-modes-run---networknonehostdefault>`_. You can use the :code:`docker_template` field to do this.
 
+First, write a :code:`Dockerfile.template` file anywhere that fits your needs.
+This template file is a mixed between a :code:`Dockerfile` and a :code:`Jinja` template file:
 
+.. code-block:: dockerfile
+
+   {% extends bento__dockerfile %}
+   {% block SETUP_BENTO_BASE_IMAGE %}
+   FROM --platform=$BUILDPLATFORM python:3.7-slim as buildstage
```

Comment:
is this line necessary? can we show a minimum example?
```suggestion
```

Reply:
maybe move the example with build stage to the reference guide. Here let's just do something simpler to demonstrate how it works. 

E.g.
```
   {% extends bento__dockerfile %}
   {% block SETUP_BENTO_COMPONENTS %}
   
   RUN echo 'We are running this during bentoml containerize!'
   
   {{ super() }}
   {% endblock %}
```